src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala

```diff
@@ -98,16 +101,63 @@ class HouseholdFleetManager(
     case GetVehicleTypes(triggerId) =>
       sender() ! VehicleTypesResponse(vehicles.values.map(_.beamVehicleType).toSet, triggerId)
 
-    case MobilityStatusInquiry(_, _, _, triggerId) =>
-      availableVehicles = availableVehicles match {
-        case firstVehicle :: rest =>
-          logger.debug("Vehicle {} is now taken", firstVehicle.id)
-          firstVehicle.becomeDriver(sender)
-          sender() ! MobilityStatusResponse(Vector(ActualVehicle(firstVehicle)), triggerId)
-          rest
-        case Nil =>
-          sender() ! MobilityStatusResponse(Vector(), triggerId)
-          Nil
+    case MobilityStatusInquiry(personId, whenWhere, _, requireVehicleCategoryAvailable, triggerId) =>
+      {
+        for {
+          neededVehicleCategory              <- requireVehicleCategoryAvailable
+          emergencyHouseholdVehicleGenerator <- maybeEmergencyHouseholdVehicleGenerator
+          vehicle                            <- emergencyHouseholdVehicleGenerator.createVehicle(personId, nextVehicleIndex, neededVehicleCategory)
+        } yield {
+          if (availableVehicles.isEmpty) {
+            // Create a vehicle out of thin air
+            nextVehicleIndex += 1
+            vehicle.setManager(Some(self))
+            vehicle.spaceTime = whenWhere
+            val mobilityRequester = sender()
+            vehiclesInternal(vehicle.id) = vehicle
+
+            // Pipe my car through the parking manager
+            // and complete initialization only when I got them all.
+            val responseFuture = parkingManager ? ParkingInquiry.init(
+              whenWhere,
+              "wherever",
+              triggerId = triggerId
+            )
+            logger.warn(
+              s"No vehicles available for category ${neededVehicleCategory} available for person ${personId.toString}, creating a new vehicle with id ${vehicle.id.toString}"
+            )
+
+            responseFuture.collect { case ParkingInquiryResponse(stall, _, otherTriggerId) =>
+              vehicle.useParkingStall(stall)
+              logger.debug("Vehicle {} is now taken, which was just created", vehicle.id)
+              vehicle.becomeDriver(mobilityRequester)
+              MobilityStatusResponse(Vector(ActualVehicle(vehicle)), otherTriggerId)
+            } pipeTo mobilityRequester
+          } else {
+            availableVehicles = availableVehicles match {
+              case firstVehicle :: rest =>
+                logger.debug("Vehicle {} is now taken", firstVehicle.id)
+                firstVehicle.becomeDriver(sender)
+                sender() ! MobilityStatusResponse(Vector(ActualVehicle(firstVehicle)), triggerId)
+                rest
+              case _ =>
```

Comment:
You can change the if to be a match and then the Nil case moves from an unnecessary logging to being the primary emergency generation. Note that you could still end up with a None vehicle in the scenario where no available vehicles and no emergency generator is configured to be available. How you want to handle that (logging?) is up to you.

Reply:
It's fine to end up with a None vehicle